agent/src/main/java/com/jonnyzzz/teamcity/plugins/node/agent/nvm/NVM.kt

```diff
@@ -105,11 +108,22 @@ class NVMRunner(val downloader : NVMDownloader,
           downloader.downloadNVM(nvmHome, url)
           message("NVM downloaded into $nvmHome")
         }
+        
         script("Install", "Installing Node.js v$version",nvmHome.path) {
-          ". $nvmHome/nvm.sh" n "nvm install $fromSource $version"
+          if (isWindows)
+            installCmd
+          else
+            ". $nvmHome/nvm.sh" n installCmd
         }
+
         script("Use", "Selecting Node.js v$version", nvmHome.path) {
-          ". $nvmHome/nvm.sh" n "nvm use $version" n "eval \${TEAMCITY_CAPTURE_ENV}"
+          if (isWindows)
+            // Executing ${TEAMCITY_CAPTURE_ENV} in Windows has no effect like it does in Linux, as agent parameters
+            // still displays TEAMCITY_CAPTURE_ENV as populated, and its value transfers between build steps without
+            // the need for an 'eval' substitute
```

Comment:
`${TEAMCITY_CAPTURE_ENV}` is an executable command line that captures all environment variables and registers those as TeamCity build environment variables. That makes it possible to use NVM configured environment in the following build steps. 

How does it work for you without it?

Reply:
Based on https://github.com/coreybutler/nvm-windows - NVM creates a symlink (NVM_HOME) to the 'current' configured Node environment. For example, if you ask NVM to install Node version 8.9.4, it installs version 8.9.4 into its own directory and then updates the symlink at NVM_HOME to point to that version. 

This means the NVM_HOME env var, always points to the same symlink, which links to the 'current' Node environment for the build process.

As the symlink at NVM_HOME changes when you inform NVM of the version of Node you wish  the path is always the same. This is why executing `${TEAMCITY_CAPTURE_ENV}` is not needed.

Have you seen any other implementations of NVM for Windows where the 'current' configured Node environment is not managed in this way?

Reply:
I did not know about that. Thank you for the clarification. The only question - do we set the NVM_HOME variable if it was not set on a build agent? Or do we expect the user to pre-set the environment variable for the whole machine? (build agent normally runs as a service and it may be tricky to have the variable included there too)

It may make sense to set up this variable explicitly for the build scope. It may also help to isolate one build from the other

Reply:
You may check an API for the AgentRunningBuild on the agent side. You have the instance in the code you've changed. Look for *parameters* in the method names